token/cli/src/main.rs

```diff
@@ -171,20 +171,35 @@ fn command_create_account(
     Ok(Some(transaction))
 }
 
-fn command_assign(config: &Config, account: Pubkey, new_owner: Pubkey) -> CommandResult {
+fn command_authorize(
+    config: &Config,
+    account: Pubkey,
+    authority_type: AuthorityType,
+    new_owner: Option<Pubkey>,
+) -> CommandResult {
+    let auth_str = match authority_type {
+        AuthorityType::MintTokens => "mint authority",
+        AuthorityType::FreezeAccount => "freeze authority",
+        AuthorityType::AccountOwner => "owner",
+        AuthorityType::CloseAccount => "close authority",
+    };
```

Comment:
It might pay to add a client-side pre-check that the authority being set is valid for the account type specified

Reply:
I considered that.
Generally, this cli tool depends on simulate transaction for a lot of those kinds of failures, and the errors it returns are a bit opaque. Depending on how much use we intend for this tool, I imagine we may want to implement more user-friendly, consistent error handling. I've been a little bit loath to add a bunch of String errors in the meantime.

Reply:
Odds are exchanges will be wrapping it.  I'm game to add checks as they become a problem though